lib/galaxy/datatypes/sniff.py

```diff
@@ -846,11 +846,6 @@ def handle_uploaded_dataset_file_internal(
         guessed_ext = ext
         if ext in AUTO_DETECT_EXTENSIONS:
             guessed_ext = guess_ext(converted_path, sniff_order=datatypes_registry.sniff_order, is_binary=is_binary)
-            guessed_datatype = datatypes_registry.get_datatype_by_extension(guessed_ext)
-            if not is_binary and guessed_datatype.is_binary:
-                # It's possible to have a datatype that is binary but not within the first 1024 bytes,
-                # so check_binary might return a false negative. This is for instance true for PDF files
-                is_binary = True
```

Comment:
As explained in the comment, we can't really trust `check_binary()` when it returns False, only when it returns True, e.g.:
```python
from galaxy.util.checkers import check_binary

check_binary("lib/galaxy/datatypes/test/454Score.pdf")
```

returns `False`. I'd keep this.

Reply:
We should improve check_binary then. There's no reason we need to spend so much time in the sniffers that are nit guaranteed to be perfect anyway.